pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test: TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr failed

pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test.TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20251677?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20251677?buildTab=artifacts#/) on master @ [5898b5ddcdff6c09e797af467424ca21f425d590](https://github.com/cockroachdb/cockroach/commits/5898b5ddcdff6c09e797af467424ca21f425d590):


```
          USE multiregion_db;
          SELECT crdb_internal.validate_multi_region_zone_configs()
          ----
    framework.go:559: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/7560/execroot/com_github_cockroachdb_cockroach/bazel-out/s390x-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:27:
          stage-exec phase=PostCommitNonRevertiblePhase stage=:
          USE multiregion_db;
          SELECT crdb_internal.validate_multi_region_zone_configs()
          ----
    panic.go:635: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/457c93847c1c7dacea53382236e46d52/logTestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr2546182851
--- FAIL: TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr (378.50s)
=== RUN   TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_5_of_24
--- FAIL: TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_5_of_24 (102.27s)
=== RUN   TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_5_of_24/restore_all_tables_in_database
    sql_runner.go:117: 
        	Error Trace:	pkg/testutils/sqlutils/sql_runner.go:130
        	            				pkg/testutils/sqlutils/sql_runner.go:391
        	            				pkg/sql/schemachanger/sctest/framework.go:964
        	            				pkg/sql/schemachanger/sctest/backup.go:464
        	Error:      	Received unexpected error:
        	            	query 'SELECT status, job_type, description FROM [SHOW JOBS] WHERE job_type IN ('SCHEMA CHANGE', 'TYPEDESC SCHEMA CHANGE', 'NEW SCHEMA CHANGE') AND status NOT IN ('succeeded', 'failed')': expected:
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry.func1
        	            	  | 	pkg/testutils/sqlutils/sql_runner.go:394
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1
        	            	  | 	pkg/testutils/soon.go:71
        	            	  | github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration
        	            	  | 	pkg/util/retry/retry.go:346
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError
        	            	  | 	pkg/testutils/soon.go:77
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).succeedsWithin
        	            	  | 	pkg/testutils/sqlutils/sql_runner.go:130
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry
        	            	  | 	pkg/testutils/sqlutils/sql_runner.go:391
        	            	  | github.com/cockroachdb/cockroach/pkg/sql/schemachanger/sctest.waitForSchemaChangesToFinish
        	            	  | 	pkg/sql/schemachanger/sctest/framework.go:964
        	            	  | github.com/cockroachdb/cockroach/pkg/sql/schemachanger/sctest.exerciseBackupRestore.func1
        	            	  | 	pkg/sql/schemachanger/sctest/backup.go:464
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1690
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_s390x.s:797
        	            	Wraps: (2) query 'SELECT status, job_type, description FROM [SHOW JOBS] WHERE job_type IN ('SCHEMA CHANGE', 'TYPEDESC SCHEMA CHANGE', 'NEW SCHEMA CHANGE') AND status NOT IN ('succeeded', 'failed')': expected:
        	            	  |
        	            	  | got:
        	            	  | running, NEW SCHEMA CHANGE, ALTER TABLE multiregion_db.public.table_regional_by_row ADD … NG HASH
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
--- FAIL: TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_5_of_24/restore_all_tables_in_database (48.99s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test.TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20252516?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20252516?buildTab=artifacts#/) on master @ [9be69f8b9cc770c7549c39aa8998a24c718f5f2a](https://github.com/cockroachdb/cockroach/commits/9be69f8b9cc770c7549c39aa8998a24c718f5f2a):


```
          USE multiregion_db;
          SELECT crdb_internal.validate_multi_region_zone_configs()
          ----
    framework.go:559: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/4853/execroot/com_github_cockroachdb_cockroach/bazel-out/s390x-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:27:
          stage-exec phase=PostCommitNonRevertiblePhase stage=:
          USE multiregion_db;
          SELECT crdb_internal.validate_multi_region_zone_configs()
          ----
    panic.go:635: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/457c93847c1c7dacea53382236e46d52/logTestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr1494100176
--- FAIL: TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr (193.30s)
=== RUN   TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_2_of_24
--- FAIL: TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_2_of_24 (161.09s)
=== RUN   TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_2_of_24/restore_all_tables_in_database
    sql_runner.go:117: 
        	Error Trace:	pkg/testutils/sqlutils/sql_runner.go:130
        	            				pkg/testutils/sqlutils/sql_runner.go:391
        	            				pkg/sql/schemachanger/sctest/framework.go:964
        	            				pkg/sql/schemachanger/sctest/backup.go:464
        	Error:      	Received unexpected error:
        	            	query 'SELECT status, job_type, description FROM [SHOW JOBS] WHERE job_type IN ('SCHEMA CHANGE', 'TYPEDESC SCHEMA CHANGE', 'NEW SCHEMA CHANGE') AND status NOT IN ('succeeded', 'failed')': expected:
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry.func1
        	            	  | 	pkg/testutils/sqlutils/sql_runner.go:394
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1
        	            	  | 	pkg/testutils/soon.go:71
        	            	  | github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration
        	            	  | 	pkg/util/retry/retry.go:346
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError
        	            	  | 	pkg/testutils/soon.go:77
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).succeedsWithin
        	            	  | 	pkg/testutils/sqlutils/sql_runner.go:130
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry
        	            	  | 	pkg/testutils/sqlutils/sql_runner.go:391
        	            	  | github.com/cockroachdb/cockroach/pkg/sql/schemachanger/sctest.waitForSchemaChangesToFinish
        	            	  | 	pkg/sql/schemachanger/sctest/framework.go:964
        	            	  | github.com/cockroachdb/cockroach/pkg/sql/schemachanger/sctest.exerciseBackupRestore.func1
        	            	  | 	pkg/sql/schemachanger/sctest/backup.go:464
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1690
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_s390x.s:797
        	            	Wraps: (2) query 'SELECT status, job_type, description FROM [SHOW JOBS] WHERE job_type IN ('SCHEMA CHANGE', 'TYPEDESC SCHEMA CHANGE', 'NEW SCHEMA CHANGE') AND status NOT IN ('succeeded', 'failed')': expected:
        	            	  |
        	            	  | got:
        	            	  | running, NEW SCHEMA CHANGE, ALTER TABLE multiregion_db.public.table_regional_by_row ADD … NG HASH
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
--- FAIL: TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_2_of_24/restore_all_tables_in_database (54.77s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test.TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20255972?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20255972?buildTab=artifacts#/) on master @ [3568bd584e6bb917debe0faf0e98bc1686750f51](https://github.com/cockroachdb/cockroach/commits/3568bd584e6bb917debe0faf0e98bc1686750f51):


```
          USE multiregion_db;
          SELECT crdb_internal.validate_multi_region_zone_configs()
          ----
    framework.go:559: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/12346/execroot/com_github_cockroachdb_cockroach/bazel-out/s390x-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:27:
          stage-exec phase=PostCommitNonRevertiblePhase stage=:
          USE multiregion_db;
          SELECT crdb_internal.validate_multi_region_zone_configs()
          ----
    panic.go:635: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/457c93847c1c7dacea53382236e46d52/logTestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr409398171
--- FAIL: TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr (287.08s)
=== RUN   TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_2_of_24
--- FAIL: TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_2_of_24 (128.01s)
=== RUN   TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_2_of_24/restore_all_tables_in_database
    sql_runner.go:117: 
        	Error Trace:	pkg/testutils/sqlutils/sql_runner.go:130
        	            				pkg/testutils/sqlutils/sql_runner.go:391
        	            				pkg/sql/schemachanger/sctest/framework.go:964
        	            				pkg/sql/schemachanger/sctest/backup.go:464
        	Error:      	Received unexpected error:
        	            	query 'SELECT status, job_type, description FROM [SHOW JOBS] WHERE job_type IN ('SCHEMA CHANGE', 'TYPEDESC SCHEMA CHANGE', 'NEW SCHEMA CHANGE') AND status NOT IN ('succeeded', 'failed')': expected:
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry.func1
        	            	  | 	pkg/testutils/sqlutils/sql_runner.go:394
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1
        	            	  | 	pkg/testutils/soon.go:71
        	            	  | github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration
        	            	  | 	pkg/util/retry/retry.go:346
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError
        	            	  | 	pkg/testutils/soon.go:77
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).succeedsWithin
        	            	  | 	pkg/testutils/sqlutils/sql_runner.go:130
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry
        	            	  | 	pkg/testutils/sqlutils/sql_runner.go:391
        	            	  | github.com/cockroachdb/cockroach/pkg/sql/schemachanger/sctest.waitForSchemaChangesToFinish
        	            	  | 	pkg/sql/schemachanger/sctest/framework.go:964
        	            	  | github.com/cockroachdb/cockroach/pkg/sql/schemachanger/sctest.exerciseBackupRestore.func1
        	            	  | 	pkg/sql/schemachanger/sctest/backup.go:464
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1690
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_s390x.s:797
        	            	Wraps: (2) query 'SELECT status, job_type, description FROM [SHOW JOBS] WHERE job_type IN ('SCHEMA CHANGE', 'TYPEDESC SCHEMA CHANGE', 'NEW SCHEMA CHANGE') AND status NOT IN ('succeeded', 'failed')': expected:
        	            	  |
        	            	  | got:
        	            	  | running, NEW SCHEMA CHANGE, ALTER TABLE multiregion_db.public.table_regional_by_row ADD … NG HASH
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
--- FAIL: TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_2_of_24/restore_all_tables_in_database (52.98s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test.TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20278318?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20278318?buildTab=artifacts#/) on master @ [e59d88e1b2e21d707124d016e8fd7373b9e972de](https://github.com/cockroachdb/cockroach/commits/e59d88e1b2e21d707124d016e8fd7373b9e972de):


```
          USE multiregion_db;
          SELECT crdb_internal.validate_multi_region_zone_configs()
          ----
    framework.go:559: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/16736/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:27:
          stage-exec phase=PostCommitNonRevertiblePhase stage=:
          USE multiregion_db;
          SELECT crdb_internal.validate_multi_region_zone_configs()
          ----
    panic.go:636: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/ee534ffac6172be5e65ec9b754536022/logTestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr4173922037
--- FAIL: TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr (211.92s)
=== RUN   TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_4_of_24
--- FAIL: TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_4_of_24 (123.59s)
=== RUN   TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_4_of_24/restore_all_tables_in_database
    sql_runner.go:117: 
        	Error Trace:	pkg/testutils/sqlutils/sql_runner.go:130
        	            				pkg/testutils/sqlutils/sql_runner.go:391
        	            				pkg/sql/schemachanger/sctest/framework.go:964
        	            				pkg/sql/schemachanger/sctest/backup.go:464
        	Error:      	Received unexpected error:
        	            	query 'SELECT status, job_type, description FROM [SHOW JOBS] WHERE job_type IN ('SCHEMA CHANGE', 'TYPEDESC SCHEMA CHANGE', 'NEW SCHEMA CHANGE') AND status NOT IN ('succeeded', 'failed')': expected:
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry.func1
        	            	  | 	pkg/testutils/sqlutils/sql_runner.go:394
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1
        	            	  | 	pkg/testutils/soon.go:71
        	            	  | github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration
        	            	  | 	pkg/util/retry/retry.go:346
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError
        	            	  | 	pkg/testutils/soon.go:77
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).succeedsWithin
        	            	  | 	pkg/testutils/sqlutils/sql_runner.go:130
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry
        	            	  | 	pkg/testutils/sqlutils/sql_runner.go:391
        	            	  | github.com/cockroachdb/cockroach/pkg/sql/schemachanger/sctest.waitForSchemaChangesToFinish
        	            	  | 	pkg/sql/schemachanger/sctest/framework.go:964
        	            	  | github.com/cockroachdb/cockroach/pkg/sql/schemachanger/sctest.exerciseBackupRestore.func1
        	            	  | 	pkg/sql/schemachanger/sctest/backup.go:464
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1792
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_arm64.s:1223
        	            	Wraps: (2) query 'SELECT status, job_type, description FROM [SHOW JOBS] WHERE job_type IN ('SCHEMA CHANGE', 'TYPEDESC SCHEMA CHANGE', 'NEW SCHEMA CHANGE') AND status NOT IN ('succeeded', 'failed')': expected:
        	            	  |
        	            	  | got:
        	            	  | running, NEW SCHEMA CHANGE, ALTER TABLE multiregion_db.public.table_regional_by_row ADD … NG HASH
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
--- FAIL: TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_4_of_24/restore_all_tables_in_database (56.47s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test.TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20290994?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20290994?buildTab=artifacts#/) on master @ [c3e379f7f513d26d305628170980f82b2d44dc0c](https://github.com/cockroachdb/cockroach/commits/c3e379f7f513d26d305628170980f82b2d44dc0c):


```
          USE multiregion_db;
          SELECT crdb_internal.validate_multi_region_zone_configs()
          ----
    framework.go:559: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2632/execroot/com_github_cockroachdb_cockroach/bazel-out/s390x-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:27:
          stage-exec phase=PostCommitNonRevertiblePhase stage=:
          USE multiregion_db;
          SELECT crdb_internal.validate_multi_region_zone_configs()
          ----
    panic.go:636: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/457c93847c1c7dacea53382236e46d52/logTestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr3287903882
--- FAIL: TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr (164.82s)
=== RUN   TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_4_of_24
--- FAIL: TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_4_of_24 (91.66s)
=== RUN   TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_4_of_24/restore_all_tables_in_database
    sql_runner.go:117: 
        	Error Trace:	pkg/testutils/sqlutils/sql_runner.go:130
        	            				pkg/testutils/sqlutils/sql_runner.go:391
        	            				pkg/sql/schemachanger/sctest/framework.go:964
        	            				pkg/sql/schemachanger/sctest/backup.go:464
        	Error:      	Received unexpected error:
        	            	query 'SELECT status, job_type, description FROM [SHOW JOBS] WHERE job_type IN ('SCHEMA CHANGE', 'TYPEDESC SCHEMA CHANGE', 'NEW SCHEMA CHANGE') AND status NOT IN ('succeeded', 'failed')': expected:
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry.func1
        	            	  | 	pkg/testutils/sqlutils/sql_runner.go:394
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1
        	            	  | 	pkg/testutils/soon.go:71
        	            	  | github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration
        	            	  | 	pkg/util/retry/retry.go:346
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError
        	            	  | 	pkg/testutils/soon.go:77
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).succeedsWithin
        	            	  | 	pkg/testutils/sqlutils/sql_runner.go:130
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry
        	            	  | 	pkg/testutils/sqlutils/sql_runner.go:391
        	            	  | github.com/cockroachdb/cockroach/pkg/sql/schemachanger/sctest.waitForSchemaChangesToFinish
        	            	  | 	pkg/sql/schemachanger/sctest/framework.go:964
        	            	  | github.com/cockroachdb/cockroach/pkg/sql/schemachanger/sctest.exerciseBackupRestore.func1
        	            	  | 	pkg/sql/schemachanger/sctest/backup.go:464
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1792
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_s390x.s:797
        	            	Wraps: (2) query 'SELECT status, job_type, description FROM [SHOW JOBS] WHERE job_type IN ('SCHEMA CHANGE', 'TYPEDESC SCHEMA CHANGE', 'NEW SCHEMA CHANGE') AND status NOT IN ('succeeded', 'failed')': expected:
        	            	  |
        	            	  | got:
        	            	  | running, NEW SCHEMA CHANGE, ALTER TABLE multiregion_db.public.table_regional_by_row ADD … NG HASH
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
--- FAIL: TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_4_of_24/restore_all_tables_in_database (50.63s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupSuccessMixedVersion_multiregion_alter_table_alter_primary_key_rbr.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/sql/schemachanger/sctest/backup.go
+++ b/pkg/sql/schemachanger/sctest/backup.go
@@ -14,6 +14,7 @@ import (
 	"strings"
 	"sync/atomic"
 	"testing"
+	"time"
 
 	"github.com/cockroachdb/cockroach/pkg/clusterversion"
 	"github.com/cockroachdb/cockroach/pkg/sql/parser"
@@ -194,6 +195,7 @@ func backupSuccess(t *testing.T, factory TestServerFactory, cs CumulativeTestCas
 	runfn := func(s serverutils.TestServerInterface, db *gosql.DB) {
 		dbForBackup.Store(db)
 		tdb := sqlutils.MakeSQLRunner(db)
+		tdb.SucceedsSoonDuration = 5 * time.Minute
 
 		// Setup the test cluster.
 		tdb.Exec(t, "CREATE DATABASE backups")

